lib/spacelift/token_source.go

```diff
@@ -0,0 +1,44 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package spacelift
+
+import "github.com/gravitational/trace"
+
+type envGetter func(key string) string
+
+// IDTokenSource allows a SpaceLift ID token to be fetched whilst within a
+// SpaceLift execution.
+type IDTokenSource struct {
+	getEnv envGetter
+}
+
+func (its *IDTokenSource) GetIDToken() (string, error) {
+	tok := its.getEnv("SPACELIFT_OIDC_TOKEN")
+	if tok == "" {
+		return "", trace.BadParameter(
+			"SPACELIFT_OIDC_TOKEN environment variable missing",
+		)
+	}
+
+	return tok, nil
+}
```

Comment:
Some places that deal with tokens try to load them from files as well.
Something like `if strings.StartsWith(tok, "/") ... os.ReadFile`

Not saying we should do this here, but have you considered?

Reply:
This variable is injected by the Spacelift Orchestrator and isn't one that's configured by the user - so I'm not sure there's any value in doing so.

Reply:
Got it :+1: